src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAnyAnalyzer.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using static Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    /// <summary>
+    /// Prefer .Length/Count/IsEmpty over Any()
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class PreferLengthCountIsEmptyOverAnyAnalyzer : DiagnosticAnalyzer
+    {
+        private const string AnyText = nameof(Enumerable.Any);
+
+        internal const string IsEmptyText = nameof(ImmutableArray<dynamic>.IsEmpty);
+        internal const string LengthText = nameof(Array.Length);
+        internal const string CountText = nameof(ICollection.Count);
+
+        internal const string IsEmptyId = "CA1860";
+        internal const string LengthId = "CA1861";
+        internal const string CountId = "CA1862";
```

Comment:
The diagnostics quite similar, do not see need to handle them separately, probably better to be one diagnostic?
CC @stephentoub @mavasani

Reply:
Resolving this comment as:
1. Other similar analyzers using different ids
2. Personally in case it was not `IEnumerable.Any()` that does count/length check in the implementation I would prefer to use Any() over Count or Length check, therefore it seems one might prefer to suppress one of them over the other.
3. With different diagnostics title and description message are cleaner

Reply:
I am unresolving this as if we restrict to `Enumerable.Any()` only there will be no need to handle these warnings separately. 

@CollinAlpert lets use only one ID for each DiagnosticDecriptors